opte/src/engine/snat.rs

```diff
@@ -37,30 +44,30 @@ cfg_if! {
 /// A single entry in the NAT pool, describing the public IP and port used to
 /// NAT a private address.
 #[derive(Clone, Copy)]
-pub struct NatPoolEntry {
-    ip: IpAddr,
+pub struct NatPoolEntry<T: ConcreteIpAddr> {
```

Comment:
Nice, this is a great way to implement this per-protocol.

Reply:
I'm 50/50 if I like this, honestly. Don't get me wrong, generics are great, I used them in places. But I'm undecided on their value in this case as you still end up with lots of duplicated code but some of it becomes a little more convoluted because of the generics. I'm obviously leaving this as-is. But just felt like giving my 2 cents.